cdap-app-fabric/src/main/java/co/cask/cdap/internal/profile/ProfileService.java

```diff
@@ -393,4 +409,23 @@ private String getCountStr(int count, String singular, String plural) {
     }
     return String.format(" and %d other %s", count - 1, plural);
   }
+
+  /**
+   * Delete the metrics for a profile.
+   *
+   * @param profileId the profile to delete metrics for.
+   */
+  private void deleteMetrics(ProfileId profileId) throws Exception {
```

Comment:
throws Exception comes from the MetricStore.delete() call right? Looking at that interface, it seems to be declaring that it throws Exception for no reason. Neither of the actual implementations throw any checked exception.

Can you do some refactoring and remove that throws clause if it's not needed? In general we should avoid anything in our internal code that throws Exception, as it makes proper error handling almost impossible.